cli.cpp

```diff
@@ -15,33 +16,58 @@ using std::string;
 using std::map;
 using std::vector;
 
+namespace {
+// Command that is being executed, used for signal handling
 ICmd *CurrentCmd;
+}
 
-size_t MaxFieldLength(std::vector<std::string> &vec, size_t min) {
+size_t MaxFieldLength(const std::vector<std::string> &vec, size_t min) {
     size_t len = 0;
-    for (auto &i : vec)
+    for (const auto &i : vec)
         if (i.length() > len)
             len  = i.length();
 
-    return (len > min ? len : min) + 2;
+    return std::max(len, min) + 2;
+}
+
+int ICmd::RunCmdImpl(const std::vector<std::string> &args,
+                     std::unique_ptr<ICmd> cmd) {
+    ICmd *prevCmd = CurrentCmd;
+
+    std::vector<char *> cargs;
+    std::vector<std::string> mutableArgs(args);
+    cargs.push_back(program_invocation_name);
+    for (auto& arg : mutableArgs)
+        cargs.push_back(&arg[0]);
+
+    cmd->SetDieOnSignal(false);
+
+    CurrentCmd = cmd.get();
+    int ret = cmd->Execute(cargs.size() - 1, (&cargs[0]) + 1);
```

Comment:
Is ICmd::Execute really need to have a non-const args?
It it possible to pass a `const char*[]` ?
Copy of args - mutableArgs can be removed in that case.